kinto/core/views/heartbeat.py

```diff
@@ -11,10 +14,35 @@ def get_heartbeat(request):
     """Return information about server health."""
     status = {}
 
+    def heartbeat_check(name, func):
+        status[name] = False
+        status[name] = func(request)
+
+    # Start executing heartbeats concurrently.
     heartbeats = request.registry.heartbeats
-    for name, callable in heartbeats.items():
-        status[name] = callable(request)
+    pool = ThreadPoolExecutor(max_workers=max(1, len(heartbeats.keys())))
+    futures = []
+    for name, func in heartbeats.items():
+        future = pool.submit(heartbeat_check, name, func)
+        future.__heartbeat_name = name  # For logging purposes.
+        futures.append(future)
+
+    # Wait for the results, with timeout.
+    seconds = float(request.registry.settings['heartbeat_timeout_seconds'])
+    done, not_done = wait(futures, timeout=seconds)
+
+    # A heartbeat is supposed to return True or False, and never raise.
+    # Just in case, go though results to spot any potential exception.
+    for future in done:
```

Comment:
Yes, I did not change the previous behaviour regarding heartbeats (the `try/except` is managed there)